mypy_django_plugin/transformers/querysets.py

```diff
@@ -619,3 +619,79 @@ def extract_prefetch_related_annotations(ctx: MethodContext, django_context: Dja
         row_type = annotated_model
 
     return default_return_type.copy_modified(args=[annotated_model, row_type])
+
+
+def _get_select_related_field_choices(model_cls: type[Model]) -> set[str]:
+    """
+    Get valid field choices for select_related lookups.
+    Based on Django's SQLCompiler.get_related_selections._get_field_choices method.
+    """
+    opts = model_cls._meta
+
+    # Direct relation fields (forward relations)
+    direct_choices = (f.name for f in opts.fields if f.is_relation)
+
+    # Reverse relation fields (backward relations with unique=True)
+    reverse_choices = (f.field.related_query_name() for f in opts.related_objects if f.field.unique)
+    return {*direct_choices, *reverse_choices}
+
+
+def _validate_select_related_lookup(
+    ctx: MethodContext,
+    django_context: DjangoContext,
+    model_cls: type[Model],
+    lookup: str,
+) -> bool:
+    """Validate a single select_related lookup string."""
+    if not lookup.strip():
+        ctx.api.fail(
+            f'Invalid field name "{lookup}" in select_related lookup. Cannot be whitespace-only',
```

Comment:
```suggestion
            f'Invalid field name "{lookup}" in select_related lookup',
```

I guess it is clear from the message.

Reply:
Fair enough